e2etest/pkg/metrics/metrics.go

```diff
@@ -31,21 +31,31 @@ type prometheusResponseData struct {
 
 // MetricsForPod returns the parsed metrics for the given pod, scraping them
 // from the source pod.
-func ForPod(controller, target *corev1.Pod, namespace string) ([]map[string]*dto.MetricFamily, error) {
+func ForPod(promPod, target *corev1.Pod, namespace string) ([]map[string]*dto.MetricFamily, error) {
```

Comment:
why did you change from controller to promPod?

Reply:
I don't remember if I had a strict reason, might be just as part of copying from what we did in frr-k8s without thinking too much if it's necessary (in frr-k8s it was). In any case, I prefer leaving this with the promPod as it's more natural to scrape from it and shouldn't change anything, wdyt?

Reply:
Yep, the reason we scrape from the controller is, originally we weren't deploying prometheus. I was curious if there was another reason behind moving other than "it's cleaner" (which is totally fine).